core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -73,6 +73,13 @@ private[spark] object Utils extends Logging {
    */
   val SPARK_CONTEXT_SHUTDOWN_PRIORITY = 50
 
+  /**
+   * The shutdown priority of temp directory must be lower than the SparkContext shutdown
+   * priority. Otherwise cleaning the temp directories while Spark jobs are running can
+   * throw undesirable errors at the time of shutdown.
+   */
+  val TEMP_DIR_SHUTDOWN_PRIORITY = 25
```

Comment:
minor nit: might be better to define this in terms of `SPARK_CONTEXT_SHUTDOWN_PRIORITY` (e.g. `SPARK_CONTEXT_SHUTDOWN_PRIORITY / 2` or `SPARK_CONTEXT_SHUTDOWN_PRIORITY - 10` or something).